relayer/shannon/config.go

```diff
@@ -0,0 +1,145 @@
+package shannon
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+)
+
+const (
+	gatewayPrivateKeyLength = 64
+	addressLength           = 43
+)
+
+var (
+	ErrShannonInvalidGatewayPrivateKey = errors.New("invalid shannon gateway private key")
+	ErrShannonInvalidGatewayAddress    = errors.New("invalid shannon gateway address")
+	ErrShannonInvalidNodeUrl           = errors.New("invalid shannon node URL")
+	ErrShannonInvalidGrpcHostPort      = errors.New("invalid shannon grpc host:port")
+)
+
+type (
+	// TODO_DISCUSS: move this (and the morse FullNodeConfig) to the config package?
+	FullNodeConfig struct {
+		RpcURL            string     `yaml:"rpc_url"`
+		GRPCConfig        GRPCConfig `yaml:"grpc_config"`
+		GatewayAddress    string     `yaml:"gateway_address"`
+		GatewayPrivateKey string     `yaml:"gateway_private_key"`
+		// TODO_UPNEXT(@adshmh): use private keys of owned apps in the configuration, and only use an app if it
+		// can be verified, i.e. if the public key derived from the stored private key matches the onchain app data.
+		// A list of addresses of onchain Applications delegated to the Gateway.
+		DelegatedApps []string `yaml:"delegated_app_addresses"`
+
+		// LazyMode, if set, will disable all caching of onchain data, specifically apps and sessions.
+		// This enables supporting short block times, e.g. when running E2E tests on LocalNet.
+		LazyMode bool `yaml:"lazy_mode"`
+	}
+
+	GRPCConfig struct {
+		HostPort          string        `yaml:"host_port"`
+		Insecure          bool          `yaml:"insecure"`
+		BackoffBaseDelay  time.Duration `yaml:"backoff_base_delay"`
+		BackoffMaxDelay   time.Duration `yaml:"backoff_max_delay"`
+		MinConnectTimeout time.Duration `yaml:"min_connect_timeout"`
+		KeepAliveTime     time.Duration `yaml:"keep_alive_time"`
+		KeepAliveTimeout  time.Duration `yaml:"keep_alive_timeout"`
+	}
+)
+
+// TODO_IMPROVE: move this to the config package?
+func (c FullNodeConfig) Validate() error {
```

Comment:
A pointer receiver is avoided here because it is not strictly required, as the `Validate()` method is not making any changes to the struct. Please let me know if I am missing something here.